docs/imagej_python.rst

```diff
@@ -11,32 +11,125 @@ This is much easier that when using the Java API and Bio-Formats plugin.
-- How to load data from OMERO
+- How to load data from OMERO.
 
-- How to run ImageJ macro from Python
+- How to run ImageJ macro from Python.
 
 Setup
 -----
 
-In this case, Fiji has been installed locally. It is possible to install Fiji "on the fly",
-see `ImageJ Tutorials <https://nbviewer.jupyter.org/github/imagej/tutorials/blob/master/notebooks/ImageJ-Tutorials-and-Demo.ipynb>`_ for more options.
+For using the Python examples and notebooks of this guide we recommend using Conda (Option 1).
+Conda manages programming environments in a manner similar to 
+`virtualenv <https://virtualenv.pypa.io/en/stable/>`_.
+
+Alternatively you can use ``repo2docker`` to build and run a Docker image locally (Option 2).
+This Docker image will provide the Conda environment and Jupyter notebooks with some image 
+analysis workflows.
+
+*Option 1*
+~~~~~~~~~~
 
 Install omero-py and `pyimagej <https://pypi.org/project/pyimagej/>`_ via Conda:
 
 - Install `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_ if necessary.
 
 - Create a programming environment using Conda and activate it::
 
-    $ conda create -n fiji_python python=3.6
+    $ conda create -n imagej_python python=3.6
 
-    $ conda activate fiji_python
+    $ conda activate imagej_python
 
 - Install omero-py and `pyimagej <https://pypi.org/project/pyimagej/>`_::
 
     $ conda install -c conda-forge pyimagej
 
     $ conda install -c ome omero-py
 
+*Option 2*
+~~~~~~~~~~
+
+Create a local Docker Image using ``repo2docker``, see :download:`README.md <https://github.com/ome/omero-guide-python/blob/master/README.md>`::
+
+    $ pip install jupyter-repo2docker
+    $ git clone https://github.com/ome/omero-guide-fiji.git
+    $ cd omero-guide-fiji
+    $ repo2docker .
+
+When the Image is ready:
+
+- Copy the URL displayed in the terminal in your favorite browser
+
+- Click the ``New`` button on the right-hand side of the window
+
+- Select ``Terminal``
+
+.. image:: images/terminal.png
+
+- A Terminal will open in a new Tab
+
+- A Conda environment has already been created when the Docker Image was built
+
+- To list all the Conda environments, run::
+
+    $ conda env list
+
+- The environment with the OMERO Python bindings and a few other libraries is named ``notebook``, activate it::
+
+    $ conda activate notebook
+
+Step-by-Step
+------------
+
+The script used in this document is :download:`run_macro_python.py <../scripts/python/run_macro_python.py>`.
+One of the advantage of this approach is that we can load only the 2D-planes we wish to analyze.
+
+In this example, Fiji has been installed locally. It is possible to install Fiji "on the fly",
+see `ImageJ Tutorials <https://nbviewer.jupyter.org/github/imagej/tutorials/blob/master/notebooks/ImageJ-Tutorials-and-Demo.ipynb>`_ for more options.
+
+To run the macro, we use ImageJ1 windows so it requires using ImageJ in GUI mode and requires handling the resulting windows. This is the reason the parameter `headless` is set to `False`.
```

Comment:
```suggestion
To run the macro, we use ImageJ1 graphical user interface (GUI) elements, and thus it requires using ImageJ in GUI mode. In this GUI mode, the resulting windows content is handled. This is the reason for the parameter `headless` to be set to `False`.
```

Reply:
I think also this paragraph needs to be nearer to the code block below where you are setting the headless parameter.

Reply:
done